src/algorithms/tracking/OrthogonalTrackSeedingConfig.h

```diff
@@ -1,102 +1,168 @@
-      m_seedFinderConfig.beamPos =
-	Acts::Vector2(m_beamPosX, m_beamPosY);
-      m_seedFinderConfig.impactMax = m_impactMax;
-      m_seedFinderConfig.rMinMiddle = m_rMinMiddle;
-      m_seedFinderConfig.rMaxMiddle = m_rMaxMiddle;
-      // Taken from SeedingOrthogonalAlgorithm.cpp, e.g.
-      // calculation of scattering using the highland formula
-      // convert pT to p once theta angle is known
-      m_seedFinderConfig.highland =
-	13.6 * std::sqrt(m_seedFinderConfig.radLengthPerSeed) *
-	(1 + 0.038 * std::log(m_seedFinderConfig.radLengthPerSeed));
-      float maxScatteringAngle =
-	m_seedFinderConfig.highland / m_seedFinderConfig.minPt;
-      m_seedFinderConfig.maxScatteringAngle2 =
-	maxScatteringAngle * maxScatteringAngle;
-
-      // Helix radius in homogeneous magnetic field
-      // in ACTS Units are GeV, mm, and GeV/(e*mm)
-      m_seedFinderConfig.pTPerHelixRadius = m_seedFinderConfig.bFieldInZ;
-      m_seedFinderConfig.minHelixDiameter2 =
-	std::pow(m_seedFinderConfig.minPt * 2 /
-		 m_seedFinderConfig.pTPerHelixRadius,
-		 2);
-
-      m_seedFinderConfig.pT2perRadius =
-	std::pow(
-          m_seedFinderConfig.highland / m_seedFinderConfig.pTPerHelixRadius,
-	  2);
-
-    }
-
-  };
-
+// Created by Dmitry Romanov
+// Subject to the terms in the LICENSE file found in the top-level directory.
+//
+
+#pragma once
+
+#include <vector>
+
+#include "Acts/Seeding/SeedFilterConfig.hpp"
+#include "Acts/Seeding/SeedFinderOrthogonalConfig.hpp"
+#include "Acts/Seeding/SeedFilter.hpp"
+#include "Acts/Seeding/SeedConfirmationRangeConfig.hpp"
+
+#include "SpacePoint.h"
+
+namespace eicrecon {
+
+  struct OrthogonalTrackSeedingConfig {
+    //////////////////////////////////////////////////////////////////////////
+    /// SEED FINDER GENERAL PARAMETERS
+    float m_rMax = 440. * Acts::UnitConstants::mm; // max r to look for hits to compose seeds
+    float m_rMin = 33. * Acts::UnitConstants::mm; // min r to look for hits to compose seeds
+    float m_zMax = 1700. * Acts::UnitConstants::mm; // max z to look for hits to compose seeds
+    float m_zMin = -1500. * Acts::UnitConstants::mm; // min z to look for hits to compose seeds
+    float m_deltaRMinTopSP = 10. * Acts::UnitConstants::mm; // Min distance in r between middle and top SP in one seed
+    float m_deltaRMaxTopSP = 200. * Acts::UnitConstants::mm; // Max distance in r between middle and top SP in one seed
+    float m_deltaRMinBottomSP = 10. * Acts::UnitConstants::mm; // Min distance in r between middle and bottom SP in one seed
+    float m_deltaRMaxBottomSP = 200. * Acts::UnitConstants::mm; // Max distance in r between middle and bottom SP in one seed
+    float m_collisionRegionMin = -250 * Acts::UnitConstants::mm; // Min z for primary vertex
+    float m_collisionRegionMax = 250 * Acts::UnitConstants::mm; // Max z for primary vertex
+
+    float m_maxSeedsPerSpM = 0; // max number of seeds a single middle sp can belong to - 1

```

Comment:
Is this a new definition in a newer version of Acts? As far as I'm aware this is just the maximum number of seeds a single middle spacepoint could belong to (not -1) - if this is true, this configuration should in principle return 0 seeds (I would naively think...) if the middle spacepoint can not belong to any triplet. Am I missing something or is this a typo?

Reply:
Thanks. Let me double check this with Emma. We definitely see seeds being generated when this parameter is set to 0, but maybe it defaults to some other value when set <1.

Reply:
Is there a chance that Emma gets back quickly on this? It would be good to include this in the release today.

Reply:
Emma is away and probably won't be able to reply until tomorrow. 

You can find the study she did for this parameter on slide 20 here:

https://indico.bnl.gov/event/18894/contributions/75735/attachments/47132/79952/param%20exploration%20%282%29.pdf

Reply:
I think it's fine to merge for now, as in any case we know from the previous study that these parameters are better than the current (wrong) ones. I think we should double check this in the future though, I'll look through the seeding code tomorrow morning to double check whether or not this is either somehow just skipped over, not representative of what I think it is, or is somehow defaulting to some other value (or none of these).

Reply:
Just to follow up on this, [here](https://github.com/acts-project/acts/blob/3b4b5c741c8541491d496a36b917b00b344d52d1/Core/include/Acts/Seeding/SeedFinderOrthogonalConfig.hpp#L49) is the definition of this parameter in Actsv21.1.0 orthogonal seeder config.

Doing a grep in my own repo for v21.1.0 I find that `maxSeedsPerSpM` is used only in `SeedFilter.ipp` [here](https://github.com/acts-project/acts/blob/3b4b5c741c8541491d496a36b917b00b344d52d1/Core/include/Acts/Seeding/SeedFilter.ipp#L270-L272). As you suggested, the algorithm makes a cut on the total number of allowed seeds+1, so the current value of 0 basically allows only 1 seed per middle spacepoint. This is probably (partly) why there are so many duplicates, especially at central rapidities, as a single space point can belong to 1 seed but the seed volume to search within encompasses >3 layers. At some point down the road it may be worth trying to optimize this value in conjunction with the seed search volume to limit the number of duplicates (while obviously also maintaining high efficiency). But in any case, my apologies for the confusion - evidently the documentation associated with the parameter itself should be updated in Acts.

Reply:
Hi Joe. Thanks for confirming this.